bin/light-base/src/runtime_service.rs

```diff
@@ -417,7 +417,7 @@ impl<TPlat: Platform> RuntimeService<TPlat> {
 
             guarded_lock.runtimes.retain(|_, rt| rt.strong_count() > 0);
 
-            if block_counts_towards_limit {
+            if !block_counts_towards_limit {
```

Comment:
don't want to bike shed, but I wouldn't call this PR "minor refactoring" as it changes the logic here. I hope it's on purpose